include/glow/Backends/TraceEvents.h

```diff
@@ -82,6 +87,52 @@ struct TraceInfo {
   }
 };
 
+/// The amount and type of TraceEvents that should appear in the trace.
+enum class TraceLevel {
+  NONE,     // No trace events.
+  RUNTIME,  // Glow runtime events only.
+  OPERATOR, // Backend operator instrumentation only.
+  STANDARD, // Glow runtime events and backend operator events.
+  DEBUG     // Full debug events with extra information.
+};
+
+/// A context for storing TraceEvents throughout a run (ie. between partitioned
+/// CompiledFunctions).
+class TraceContext {
+  std::vector<TraceEvent> traceEvents_;
+  TraceLevel traceLevel_{TraceLevel::NONE};
+  int traceThread_{0};
```

Comment:
Whats this? Can you add a comment

Reply:
This is the tid in the output trace, ie the row in the final visualization. In the examples so far we have a tid per DeviceManager but we may want to use a different scheme once we get some experience working with traces.

I'll add a comment to this effect.